Enable to initialize spatial/temporal/angular chirped laser + CI test (#1136)
Hi-PACE/hipace: docs/source/run/parameters.rst

* ``<laser name>.propagation_angle_yz`` (`float`) optinal (default `0`)
    Propagation angle of the pulse in the yz plane (0 is the along the z axis)

* ``<laser name>.chirp_theta_xy`` (`float`) optional (default `pi/2`)
    Direction of the linear spatial and angular chirp on x-y plane.

* ``<laser name>.beta`` (`float`) optional (default `0.`)
    Angular dispersion (or angular chirp) at focus defined by `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.

* ``<laser name>.zeta`` (`float`) optional (default `0.`)
    Spatial chirp at focus defined by `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.

* ``<laser name>.phi2`` (`float`) optional (default `pi/2`)
    The amount of temporal chirp :math:`\phi^{(2)}` at focus (in the lab frame).
    Namely, a wave packetcentered on the frequency :math:`(\omega_0 + \delta \omega)` will reach its peak intensity at :math:`z(\delta \omega) = z_0 - c \phi^{(2)} \, \delta \omega`.
    Thus, a positive :math:`\phi^{(2)}` corresponds to positive chirp, i.e. red part of the spectrum in the front of the pulse and blue part of the spectrum in the back.

>     More specifically, the electric field in the focal plane is of the form:

Reviewer: ```suggestion
* ``<laser name>.phi2`` (`float`) optional (default `pi/2`)
    The amount of temporal chirp :math:`\phi^{(2)}` at focus (in the lab frame).
    Namely, a wave packet centered on the frequency :math:`(\omega_0 + \delta \omega)` will reach its peak intensity at :math:`z(\delta \omega) = z_0 - c \phi^{(2)} \, \delta \omega`.
    Thus, a positive :math:`\phi^{(2)}` corresponds to positive chirp, i.e., red part of the spectrum in the front of the pulse and blue part of the spectrum in the back.
    More specifically, the electric field in the focal plane is of the form:
```

* ``<laser name>.propagation_angle_yz`` (`float`) optinal (default `0`)
    Propagation angle of the pulse in the yz plane (0 is the along the z axis)

* ``<laser name>.chirp_theta_xy`` (`float`) optinal (default `pi/2`)

>     Direction of the linear spatial and angular chirp on XoY plane.

Reviewer: ```suggestion
* ``<laser name>.chirp_theta_xy`` (`float`) optional (default `pi/2`)
    Direction of the linear spatial and angular chirp on x-y plane.
```

* ``<laser name>.beta`` (`float`) optional (default `0.`)
    Angular dispersion (or angular chirp) at focus defined by `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.

* ``<laser name>.zeta`` (`float`) optional (default `0.`)
    Spatial chirp at focus defined by `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.

* ``<laser name>.phi2`` (`float`) optional (default `pi/2`)
    The amount of temporal chirp :math:`\phi^{(2)}` at focus (in the lab frame). Namely, a wave packet
    centered on the frequency :math:`(\omega_0 + \delta \omega)` will reach its peak intensity
    at :math:`z(\delta \omega) = z_0 - c \phi^{(2)} \, \delta \omega`. Thus, a positive
    :math:`\phi^{(2)}` corresponds to positive chirp, i.e. red part of the spectrum in the
    front of the pulse and blue part of the spectrum in the back. More specifically, the electric

>     field in the focal plane is of the form:

Reviewer: For the doc, 1 sentence per line is OK, even if it's > 100 char. Otherwise, it's a bit of a pain to change it, and doesn't affect the rendering anyway.

          Distance at which the laser pulse is focused (in the z direction, counted from laser initial position).

      * ``<laser name>.propagation_angle_yz`` (`float`) optional (default `0`)

>           Propagation angle of the pulse in the yz plane (0 is along z axis)

Reviewer: ```suggestion
          Propagation angle of the pulse in the yz plane (0 is along the z axis)
```

      * ``<laser name>.STC_theta_xy`` (`float`) optional (default `pi/2`)

>           Direction of the linear spatial and angular chirp on xoy plane (0 is along x axis).

Reviewer: ```suggestion
      * ``<laser name>.STC_theta_xy`` (`float`) optional (default `0`)
          Direction of the linear spatial and angular chirps in the xy plane (`0` is along x, `pi/2` along y).
          In what follows, all chirps are given as defined in `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.
```
to be consistent with notations at line 904. IMO default 0 (i.e. x axis) is more intuitive.

      * ``<laser name>.beta`` (`float`) optional (default `0.`)
          Angular dispersion (or angular chirp) at focus defined by `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.

      * ``<laser name>.zeta`` (`float`) optional (default `0.`)

>           Spatial chirp at focus defined by `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.

Reviewer: Also, could you specify units?
```suggestion
      * ``<laser name>.beta`` (`float`) optional (default `0.`)
          Angular dispersion (or angular chirp) at focus.

      * ``<laser name>.zeta`` (`float`) optional (default `0.`)
          Spatial chirp at focus.
```

      * ``<laser name>.phi2`` (`float`) optional (default `pi/2`)

>           The amount of temporal chirp :math:`\phi^{(2)}` at focus (in the lab frame).

Reviewer: To make style consistent. Also, could add units. Finally, "in the lab frame" does not really apply here.
It is nice to write the expression `:math:`\phi^{(2)}``, but could this be made consistent also with the parameters above? I understand this one is a bit different, as there are 2 phi_2, but still.
```suggestion
          Temporal chirp :math:`\phi^{(2)}` at focus.
```

>           Namely, a wave packet centered on the frequency :math:`(\omega_0 + \delta \omega)` will reach its peak intensity at :math:`z(\delta \omega) = z_0 - c \phi^{(2)} \, \delta \omega`.
>           Thus, a positive :math:`\phi^{(2)}` corresponds to positive chirp, i.e., red part of the spectrum in the front of the pulse and blue part of the spectrum in the back.
>           More specifically, the electric field in the focal plane is of the form:

Reviewer: That could be co-authored by FBPIC people, in that case Rémi I suspect.
```suggestion
          Namely, a wave packet centered on frequency :math:`(\omega_0 + \delta \omega)` reaches its peak intensity at :math:`z(\delta \omega) = z_0 - c \phi^{(2)} \, \delta \omega`.
          Thus, a positive :math:`\phi^{(2)}` corresponds to positive chirp, i.e., red part of the spectrum in the front of the pulse and blue part in the back.
          More specifically, the electric field in the focal plane is of the form:
```

          .. math::

              E(\boldsymbol{x},t) \propto Re\left[ \exp\left(  -\frac{(t-t_{peak})^2}{\tau^2 + 2i\phi^{(2)}} + i\omega_0 (t-t_{peak}) + i\phi_0 \right) \right]

          where :math:`\tau` is given by ``<laser_name>.tau`` and represents the Fourier-limited duration of the laser pulse. Thus, the actual duration of the chirped laser pulse is:

          .. math::

               \tau' = \sqrt{ \tau^2 + 4 (\phi^{(2)})^2/\tau^2 }

>          See also the definition in `S. Akturk et al., Optics Express 12, 4399 (2004) <https://doi.org/10.1364/OPEX.12.004399>`__.
> 

Reviewer: Can be removed if there's 1 clear mention of Akturk above. Let's check that it looks good in the final documentation.
```suggestion
```

      * ``<laser name>.propagation_angle_yz`` (`float`) optional (default `0`)
          Propagation angle of the pulse in the yz plane (0 is along the z axis)

      * ``<laser name>.tau`` (`float`) optional (default `0`)
          The laser pulse duration. The pulse length is set to `laser.tau`:math:`*c_0`.
          Use either the pulse length or the pulse duration.

      * ``<laser name>.focal_distance`` (`float`)

>           Distance at which the laser pulse if focused (in the z direction, counted from laser initial position).

Reviewer: Bug in merge conflict resolution + typos.
